audio_reader.py

```diff
@@ -0,0 +1,110 @@
+import fnmatch
+import os
+import re
+import threading
+
+import librosa
+import numpy as np
+import tensorflow as tf
+
+
+def find_files(directory, pattern='*.wav'):
+    '''Recursively finds all files matching the pattern.'''
+    files = []
+    for root, dirnames, filenames in os.walk(directory):
+        for filename in fnmatch.filter(filenames, pattern):
+            files.append(os.path.join(root, filename))
+    return files
+
+
+def load_generic_audio(directory, sample_rate):
+    '''Generator that yields audio waveforms from the directory.'''
+    files = find_files(directory)
+    for f in files:
+        audio, sr = librosa.load(f, sr=sample_rate, mono=True)
+        audio = audio.reshape(-1, 1)
+        yield audio
+
+
+def load_vctk_audio(directory, sample_rate):
+    '''Generator that yields audio waveforms from the VCTK dataset, and
+    additionally the ID of the corresponding speaker.'''
+    files = find_files(directory)
+    speaker_re = re.compile(r'p([0-9]+)_([0-9]+)\.wav')
+    for filename in files:
+        audio, sr = librosa.load(filename, sr=sample_rate, mono=True)
+        audio = audio.reshape(-1, 1)
+        matches = speaker_re.findall(filename)[0]
+        speaker_id, recording_id = [int(id_) for id_ in matches]
+        yield audio, speaker_id
+
+
+def trim_sample(audio, threshold=0.3):
```

Comment:
That's a good idea and and wouldn't be too difficult to implement.
The current implementation sounded fine when inspecting the concatenated audio, but it can certainly be improved.